bootstrap.js

```diff
@@ -0,0 +1,58 @@
+/* eslint-disable import/no-dynamic-require,global-require */
+const fs = require('fs');
+const path = require('path');
+const spawn = require('child_process').spawnSync;
+
+const type = process.env.type || 'upgrade'; // yarn update types
+const logTask = msg => console.log(`👍  ${msg}`);
+
+const installPackage = (at) => {
+  const result = spawn('yarn', [type], { stdio: 'inherit', cwd: at });
+  if (result.error) {
+    console.log(result.error);
+    process.exit(1);
+  }
+  logTask(`installed ${at}\n`);
+};
+
+const packages = fs.readdirSync('packages').reduce((pkgs, pkg) => {
+  const packagePath = path.join(process.cwd(), 'packages', pkg);
+  const packageJSON = path.join(packagePath, 'package.json');
+  try {
+    if (fs.statSync(packagePath).isDirectory() && fs.statSync(packageJSON).isFile()) {
+      pkgs.push({ path: packagePath, name: require(packageJSON).name });
+    }
+  } catch (e) { return pkgs; }
+  return pkgs;
+}, []);
+
+console.log(`\n🔥  Bootstrapping\n`);
+
+// Install the root package.json first so that we can use shelljs.
+installPackage(process.cwd());
+const shell = require('shelljs');
+
+// Install all of the monorepo packages.
+packages.forEach(pkg => installPackage(pkg.path));
+
+// Symlink monorepo package dependencies to local packages.
+packages.forEach((pkg) => {
```

Comment:
so the assumption is that the last package which contains dependency of said version will be the chosen version as it was the last symlinked? Wondering if this has caused or might cause interdependency related errors in the past/future? Do you have a recommended approach to ensuring the dependencies as you say as "latest published versions"? do you suggest to use `@latest` for example? Or how do you sanity check this in terms of upgrading workflow?  Should this code emit warnings to handle this case for example?